reliability/single-fit/trace/src/test/kotlin/com/datadog/android/trace/integration/otel/TracerExt.kt

```diff
@@ -6,21 +6,23 @@
 
 package com.datadog.android.trace.integration.otel
 
+import com.datadog.android.api.feature.Feature
+import com.datadog.android.api.feature.FeatureScope
 import com.datadog.android.trace.integration.tests.utils.BlockingWriterWrapper
 import com.datadog.tools.unit.getFieldValue
 import com.datadog.tools.unit.setFieldValue
 import com.datadog.trace.common.writer.Writer
-import com.datadog.trace.core.CoreTracer
-import io.opentelemetry.api.trace.Tracer
 
-internal fun Tracer.useBlockingWriter(): BlockingWriterWrapper {
-    val coreTracer: CoreTracer = this.getFieldValue("tracer")
-    val writer: Writer = coreTracer.getFieldValue("writer")
-    return if (writer !is BlockingWriterWrapper) {
+private const val WRITER_FIELD_NAME = "coreTracerDataWriter"
+
+internal fun FeatureScope?.useBlockingWriter(): BlockingWriterWrapper {
+    val feature = this?.unwrap<Feature>() ?: throw NullPointerException("Can't cast to Feature")
```

Comment:
minor: normally cast to `Feature` should always succeed, because the method is declared as `fun <T : Feature> unwrap(): T`. If it doesn't succeed, it will throw `ClassCastException`.

the only way to get `null` here is to have `null` receiver.